### PR TITLE
Adds TRUSTED_DIR to maintain the same pattern as sgx_u.mk.

### PR DESCRIPTION
**Pull request by Icaro-Lima**
e.g.:
`static_trusted/$(enclaveName)_t.h: ...`
Changed to:
`$(TRUSTED_DIR)/$(enclaveName)_t.h: ...`

### CODE REVIEW

**Comment by xiangquanliu** [comment]
If you'd like to do this change, please update these files accordingly. Moreover, please also add the macro "TRUSTED_DIR" in the untrusted make file.
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXEnclave/makefiles/c++/sgx_t.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXEnclave/makefiles/c++/sgx_u.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXEnclave/makefiles/c++/sgx_u.without_app.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXEnclave/makefiles/c/sgx_t.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXEnclave/makefiles/c/sgx_u.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXEnclave/makefiles/c/sgx_u.without_app.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXTrustedStaticLibrary/makefiles/c++/sgx_t_static.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXTrustedStaticLibrary/makefiles/c++/sgx_u.without_app.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXTrustedStaticLibrary/makefiles/c/sgx_t_static.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXTrustedStaticLibrary/makefiles/c/sgx_u.without_app.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXUntrustedModule/makefiles/c++/sgx_u.without_app.mk
Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXUntrustedModule/makefiles/c/sgx_u.without_app.mk

**Comment by Icaro-Lima** [comment]
Okay, I'm already doing it.

File: Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXEnclave/makefiles/c++/sgx_t.mk [hunk 1]
```diff
@@ -100,28 +101,28 @@ endif
 
 ######## $(enclaveName) Objects ########
 
-trusted/$(enclaveName)_t.c: $(SGX_EDGER8R) ./trusted/$(enclaveName).edl
-	@cd ./trusted && $(SGX_EDGER8R) --trusted ../trusted/$(enclaveName).edl --search-path ../trusted --search-path $(SGX_SDK)/include
+$(TRUSTED_DIR)/$(enclaveName)_t.c: $(SGX_EDGER8R) ./$(TRUSTED_DIR)/$(enclaveName).edl
+	@cd ./$(TRUSTED_DIR) && $(SGX_EDGER8R) --$(TRUSTED_DIR) ../$(TRUSTED_DIR)/$(enclaveName).edl --search-path ../$(TRUSTED_DIR) --search-path $(SGX_SDK)/include
```
Comment (Icaro-Lima) [inline Linux_SGXEclipsePlugin/build_directory/plugins/com.intel.sgx/templates/sgx/SGXEnclave/makefiles/c++/sgx_t.mk]: I fixed this on the next branch.

**Comment by Icaro-Lima** [comment]
Done.

**Comment by Icaro-Lima** [comment]
Done.

**Comment by lzha101** [comment]
@Icaro-Lima, before we merge the PR, could you please help to update the patch to resolve the conflicts with the latest release?

**Comment by Icaro-Lima** [comment]
Done.

**Pull request merged**